kafka-rest/src/main/java/io/confluent/kafkarest/entities/GetTopicPartitionOffsetResponse.java

```diff
@@ -0,0 +1,143 @@
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.entities;
+
+import static java.util.Objects.requireNonNull;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonProperty.Access;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/**
+ * Response for GET /topics/(topic)/partitions/(partition)/offsets requests.
+ *
+ * <p>It will either be:</p>
+ *
+ * <ul>
+ *   <li>a summary containing the beginning and end offsets</li>
+ *   <li>an offset and its associated timestamp</li>
+ *   <li>empty</li>
+ * </ul>
+ *
+ * @see io.confluent.kafkarest.resources.v2.PartitionsResource#getOffsets(String, int, Instant)
+ */
+public final class GetTopicPartitionOffsetResponse {
+
+  private static final GetTopicPartitionOffsetResponse EMPTY =
+      new GetTopicPartitionOffsetResponse(
+          /* beginning= */ null, /* end= */ null, /* offset= */ null, /* timestamp= */ null);
+
+  private final @Nullable Long beginning;
+  private final @Nullable Long end;
+  private final @Nullable Long offset;
+  private final @Nullable Instant timestamp;
+
+  private GetTopicPartitionOffsetResponse(
+      @Nullable Long beginning,
+      @Nullable Long end,
+      @Nullable Long offset,
+      @Nullable Instant timestamp
+  ) {
+    this.beginning = beginning;
+    this.end = end;
+    this.offset = offset;
+    this.timestamp = timestamp;
+  }
+
+  /**
+   * Returns a response with a summary containing the {@code beginning} and {@code end} offsets.
+   */
+  public static GetTopicPartitionOffsetResponse createSummary(long beginning, long end) {
+    return new GetTopicPartitionOffsetResponse(
+        beginning, end, /* offset= */ null, /* timestamp= */ null);
+  }
+
+  /**
+   * Returns a response with the given {@code offset} and its associated {@code timestamp}.
+   */
+  public static GetTopicPartitionOffsetResponse createForTime(long offset, Instant timestamp) {
+    return new GetTopicPartitionOffsetResponse(
+        /* beginning= */ null, /* end= */ null, offset, requireNonNull(timestamp, "timestamp"));
+  }
+
+  /**
+   * Returns an empty response.
+   */
+  public static GetTopicPartitionOffsetResponse empty() {
+    return EMPTY;
+  }
+
+  /**
+   * The earliest offset in the topic partition.
+   */
+  @JsonProperty(value = "beginning", access = Access.READ_ONLY)
```

Comment:
i'm personally inclined towards the more verbose `beginning_offset` and `end_offset`. I agree with the choice of words `beginning` and `end` as they are consistent with the java api (i would choose other words if that wasn't a constraint though).